File: app/src/main/java/com/jbproductions/liszt/ListDao.java
package com.jbproductions.liszt;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

/**
 * DAO to provide query functionality for TaskList objects.
 */
@Dao
public interface ListDao {

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    void insert(TaskList list);

    @Update
    void updateList(TaskList list);

    @Query("SELECT * FROM lists WHERE id = :id")
    TaskList getListById(long id);

}

File: app/src/main/java/com/jbproductions/liszt/dao/ListDao.kt
package com.jbproductions.liszt.dao

import androidx.room.*
import com.jbproductions.liszt.models.ListModel

/**
 * DAO to provide query functionality for ListModel objects.
 */
@Dao
interface ListDao {
    @Insert(onConflict = OnConflictStrategy.IGNORE)
    suspend fun insert(list: ListModel)

    @Update
    suspend fun updateList(list: ListModel)

    @Query("SELECT * FROM lists WHERE id = :id")
    suspend fun getListById(id: Long): ListModel?
}